pyproject.toml

```diff
@@ -3,13 +3,75 @@
 # Note that these settings are not respected with pre-commit run --all-files
 # In that case add configurations to the .pre-commit-config.yaml file.
 
-[project]
+[tool.poetry]
```

Comment:
Mmm, so we are going all in for poetry, huh?
Cool

I was holding it back as an idea for a long while now. But yeah, with optimizing container size it's a good time to start using it.

I'll add some comments regarding it then.